nixos/modules/services/web-servers/caddy/default.nix

```
@@ -267,6 +273,16 @@ in
       '';
     };
 
+    enableReload = mkOption {
+      default = false;
+      type = types.bool;
+      description = lib.mdDoc ''
+        Reload Caddy when configuration file changes (instead of restart).
+        The configuration file is exposed at {file}`/etc/${etcConfigFile}`.
+        If you set this option, consider setting a grace_period value in the
+        global config to prevent Caddy from hanging up indefinitely.
```

Bit picky, but how about
```suggestion
        Reload Caddy when configuration file changes (instead of restart).
        The configuration file is exposed at {file}`/etc/${etcConfigFile}`.
        If you enable this option, consider setting [`grace_period`](https://caddyserver.com/docs/caddyfile/options#grace-period)
        to a non-infinite value in {option}`services.caddy.globalConfig`
        to prevent Caddy waiting for active connections to finish,
        which could delay the reload essentially indefinitely.
```